x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/operator/exchange/ExchangeSink.java

```diff
@@ -40,4 +40,9 @@ public interface ExchangeSink {
      * Whether the sink is blocked on adding more pages
      */
     IsBlockedResult waitForWriting();
+
+    /**
+     * Session ID for this exchange
+     */
+    String exchangeId();
```

Comment:
I am confused about this one. If it is session id, isn't it the same for the entire query?

Reply:
When the compute engine creates sub-tasks, it seems to create child-session-ids by appending suffixes to the original session-id. These are also called sessionIds in some code (and externalId and exchangeId in other places). I have not made complete sense of it all yet, but can certainly report that I did see more than one sessionId in tests (although all had the same suffix).